lib/core/src/Cardano/Wallet/Binary.hs

```diff
@@ -586,8 +589,8 @@ encodeTxOut (TxOut (Address addr) (Coin c)) = mempty
 -- It returns an hex-encoded 64-byte hash.
 --
 -- NOTE: This is a rather expensive operation
-txId :: Tx -> Hash "Tx"
-txId = blake2b256 . encodeTx
+instance TxId HttpBridge where
+    txId = blake2b256 . encodeTx
```

Comment:
so my understanding is that txId implemented by HttpBridge (ie, rust code) is something not strictly specified and is likeley to change in the future

Reply:
its pretty late so I might be wrong.

I see two scenarios that might be motivation for introduction of more flexible txId:
 1) One day we might need to change representation of txId and we will need to switch. When we switch , we will probably have to support old and new version
 2) Maybe there is a need to support multiple ways how to compute txId at the same time

for 2) it totally makes sense to implement it - if there is a need to support multiple versions at the same time. If 1) is expected to happen rather soonish, then added complexity makes sense.

(this is my attempt to figure out why we needed this feature - the explanations above, if true, are enough for me)

Reply:
@akegalj So yes, sorry it's coming a bit of nowhere (more on Monday in our planning meeting); We will, in the very near future, start integrating with the Rust nodes which use a different binary format for blocks, addresses and transactions. With also in some cases, different hashing algorithms. On the other hand, we do want to keep open this current implementation with Byron and the bridge for the transition period and also because, that's what the Haskell nodes will also be using initially (with maybe some small variations, so we may likely end up with 3 different instances of the above!)